app/models/job_application.rb

```diff
@@ -44,4 +45,35 @@ class JobApplication < ApplicationRecord
   def update_status_timestamp
     self["#{status}_at"] = Time.current
   end
+
+  def anonymise_report
+    return unless status == "submitted"
+
+    fill_in_report
+    anonymise_equal_opportunities_attributes
+  end
+
+  def fill_in_report
+    report = vacancy.equal_opportunities_report || vacancy.build_equal_opportunities_report
+    Jobseekers::JobApplication::EqualOpportunitiesForm::ATTRIBUTES.each do |attr|
+      attr_value = public_send(attr)
+      if attr.ends_with?("_description")
+        next unless attr_value.present?
+
+        attr_name = attr.to_s.split("_").first
+        report.public_send("#{attr_name}_other_descriptions") << attr_value
+      else
+        report.increment("#{attr}_#{attr_value}")
+      end
+    end
+    report.increment(:total_submissions)
+    report.save
+  end
+
+  def anonymise_equal_opportunities_attributes
+    Jobseekers::JobApplication::EqualOpportunitiesForm::ATTRIBUTES.each do |attr|
+      attr_value = public_send(attr)
+      self[attr] = StringAnonymiser.new(attr_value)
```

Comment:
When we say "anonymise" here, what we want to do is set these to `nil` – it would be too easy to reconstruct if we just run them through the `StringAnonymiser` hashing (because there are only a handful of options to begin with).